### PR TITLE
feat(modules.clickhouse): Add zookeeper for clickhouse clusterization

### PR DESCRIPTION
**Pull request by laskoviymishka**
If you would like to perform queries against clusterized ClickHouse you need to start zookeeper near your cluster and pass link to it via `config.xml`

API example:

```
	zkhost, zkport, err := RunZookeeper(ctx)
	if err != nil {
		t.Fatal(err)
	}

	container, err := RunContainer(ctx,
		WithUsername(user),
		WithPassword(password),
		WithDatabase(dbname),
		WithZookeeper(zkhost, zkport),
	)
	if err != nil {
		t.Fatal(err)
	}
```

## What does this PR do?

Add option: `WithZookeeper` to clickhouse module.

## Why is it important?

ClickHouse queries (especially DDLs) vary a lot between single node and cluster. Having it a clusterized allows you to apply exactly same code to real prod clusters and in your tests.

### CODE REVIEW

File: modules/clickhouse/zookeeper.go
Comment (eddumelendez) [inline modules/clickhouse/zookeeper.go:104]: I don't think the clickhouse module should have a zookeeper implementation but rather just receive the connection string to be used in clickhouse

File: modules/clickhouse/zookeeper.go
Comment (eddumelendez) [inline modules/clickhouse/zookeeper.go:78]: this would be a limitation for those who need other configurations.

**Review by eddumelendez — COMMENTED** [review]
Thanks for your contribution. I've left some comments but in general I would rather prefer to add an example about how to do this instead of hardcoding the configuration file. Also, wonder if it is possible to configure the zookeeper section to do not assume other config values. If so, then using templates would be great, redpanda module is doing so [here](https://github.com/testcontainers/testcontainers-go/tree/main/modules/redpanda/mounts).

With Testcontainers, containers can be composed programmatically and with the upcoming network support in https://github.com/testcontainers/testcontainers-go/pull/1993 modules would improve this experience. That's why I don't see having zookeeper as part of clickhouse.

File: modules/clickhouse/zookeeper.go
Comment (laskoviymishka) [reply modules/clickhouse/zookeeper.go:104]: I can move zk container into a separate module, but it's like 20 lines of code, not sure regards it worth a hassle.

Comment (laskoviymishka) [reply modules/clickhouse/zookeeper.go:104]: I can move this code just in examples

File: modules/clickhouse/zookeeper.go
Comment (laskoviymishka) [reply modules/clickhouse/zookeeper.go:78]: actually, it's possible - https://clickhouse.com/docs/en/operations/configuration-files
I will put this zookeeper config in a separate file.

File: modules/clickhouse/mounts/zk_config.xml.tpl
Comment (eddumelendez) [inline modules/clickhouse/mounts/zk_config.xml.tpl:30]: are those default values? Can we just keep the zookeeper section?

File: modules/clickhouse/clickhouse.go
Comment (eddumelendez) [inline modules/clickhouse/clickhouse.go:93]:
```suggestion
		return nil, fmt.Errorf("failed to parse zookeeper config file template: %w", err)
```

File: modules/clickhouse/clickhouse.go
Comment (eddumelendez) [inline modules/clickhouse/clickhouse.go:98]:
```suggestion
		return nil, fmt.Errorf("failed to render zookeeper bootstrap config template: %w", err)
```

File: docs/modules/clickhouse.md
Comment (eddumelendez) [inline docs/modules/clickhouse.md:78]:
```suggestion
Clusterized ClickHouse requires to start Zookeeper and pass link to it via `config.xml`.
```

File: modules/clickhouse/mounts/zk_config.xml.tpl
Comment (laskoviymishka) [reply modules/clickhouse/mounts/zk_config.xml.tpl:30]: all this is a part of zookeeper default configuration, without it cluster still considered as not replicated one.

**Comment by laskoviymishka** [comment]
@eddumelendez gentle ping on this one <3

**Review by eddumelendez — DISMISSED** [review]
LGTM

File: modules/clickhouse/clickhouse.go
Comment (mdelapenya) [inline modules/clickhouse/clickhouse.go:107]: I think we should document that this function could panic under this condition. Would you mind adding it to the docs?

File: modules/clickhouse/clickhouse.go
Comment (mdelapenya) [inline modules/clickhouse/clickhouse.go:118]: Same here

**Review by mdelapenya — COMMENTED** [review]
LGTM, will approve it once the comments for the docs are addressed.

Thank you so much for your work here! And @eddumelendez, thanks for the review too!

File: modules/clickhouse/clickhouse.go
Comment (mdelapenya) [reply modules/clickhouse/clickhouse.go:107]: I went ahead and added the docs 🙏  Once the CI passes, I think we can merge this one. Thanks!

File: modules/clickhouse/clickhouse_test.go
Comment (mdelapenya) [inline modules/clickhouse/clickhouse_test.go:251]: I'm about to merge this one, but @eddumelendez: how would you see if this `WithZookeeper` option automatically starts the container for you? Would this be overkill? Or is it preferred this way, that allows setting up an external zookeeper passing the IP/port to the clickhouse container? For the "embedded" one, it can be done as a follow-up PR

Comment (mdelapenya) [reply modules/clickhouse/clickhouse_test.go:251]: As discussed internally, having the `WithZookeper` options could be implemented, but it's not a priority at all, as the supporting containers (the Zookeeper) can be managed from outside the module.

**Comment by mdelapenya** [comment]
Thanks @laskoviymishka for submitting this new feature! Much appreciated 🙇

**Pull request merged**
